### PR TITLE
Update query service and improve SQL backend

### LINKED ISSUE #843: Add code to initialize and/or reset the Postgres database when using SQL persistence
Given a database name and credentials for a user with permissions over that database, when creating a `SqlDataSource` we should
* Remove all existing tables if requested (`--reset-store`)
* Run migrations to create necessary tables

### PR DESCRIPTION
**Pull request by jbearer**
* Use new Query API for database queries
* Make database resettable
* Fix some tests, which were failing based on timing

Closes #843

### CODE REVIEW

File: sequencer/src/api/options.rs [hunk 1]
```diff
@@ -184,6 +184,10 @@ pub struct Sql {
     /// Password for Postgres user.
     #[clap(long, env = "ESPRESSO_SEQUENCER_POSTGRES_PASSWORD")]
     pub password: Option<String>,
+
+    /// Reset database upon connecting.
+    #[clap(long, env = "ESPRESSO_SEQUENCER_RESET_STORE")]
```
Comment (sveitser) [inline sequencer/src/api/options.rs:189]: This is a bit annoying because it only supports `true` (not `1`, `True`, ...).

Comment (sveitser) [reply sequencer/src/api/options.rs:189]: @Ancient123 what's your preferred way to use this feature?

**Review by sveitser — APPROVED** [review]
LGTM. A question about UX.

File: sequencer/src/api/options.rs (hunk 1 above)
Comment (Ancient123) [reply sequencer/src/api/options.rs:189]: true is fine, but supporting more options would be better.. Not worth holding up the PR though honestly.

**Pull request merged**